components/camera/replaypcd/replaypcd_utils_test.go

```diff
@@ -0,0 +1,184 @@
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var dataBuf bytes.Buffer
+	gz := gzip.NewWriter(&dataBuf)
+	gz.Write(data)
+	gz.Close()
+
+	// Construct response
+	binaryData := &datapb.BinaryData{
+		Binary:   dataBuf.Bytes(),
+		Metadata: &datapb.BinaryMetadata{},
+	}
+
+	resp := &datapb.BinaryDataByFilterResponse{
+		Data: []*datapb.BinaryData{binaryData},
+		Last: fmt.Sprint(newFileNum),
+	}
+	return resp, nil
+}
+
+// createMockCloudDependencies creates a mockDataServiceServer and rpc client connection to it which is then
+// stored in a mockCloudConnectionService.
+func createMockCloudDependencies(ctx context.Context, t *testing.T, logger golog.Logger, b bool) (resource.Dependencies, func() error) {
+	listener, err := net.Listen("tcp", "localhost:0")
+	test.That(t, err, test.ShouldBeNil)
+	rpcServer, err := rpc.NewServer(logger, rpc.WithUnauthenticated())
+	test.That(t, err, test.ShouldBeNil)
+
+	test.That(t, rpcServer.RegisterServiceServer(
+		ctx,
+		&datapb.DataService_ServiceDesc,
+		&mockDataServiceServer{},
+		datapb.RegisterDataServiceHandlerFromEndpoint,
+	), test.ShouldBeNil)
+
+	go rpcServer.Serve(listener)
+
+	conn, err := viamgrpc.Dial(ctx, listener.Addr().String(), logger)
+	test.That(t, err, test.ShouldBeNil)
+
+	mockCloudConnectionService := &cloudinject.CloudConnectionService{
+		Named: cloud.InternalServiceName.AsNamed(),
+		Conn:  conn,
+	}
+	if !b {
+		mockCloudConnectionService.AcquireConnectionErr = errors.New("cloud connection error")
+	}
+
+	r := &inject.Robot{}
+	rs := map[resource.Name]resource.Resource{}
+	rs[cloud.InternalServiceName] = mockCloudConnectionService
+	r.MockResourcesFromMap(rs)
+
+	return resourcesFromDeps(t, r, []string{cloud.InternalServiceName.String()}), rpcServer.Stop
+}
+
+// createNewReplayPCDCamera will create a new replay_pcd camera based on the provided config with either
+// a valid or invalid data client.
+func createNewReplayPCDCamera(ctx context.Context, t *testing.T, replayCamCfg *Config, validDeps bool,
+) (camera.Camera, func() error, error) {
+	logger := golog.NewTestLogger(t)
+
+	resources, closeRPCFunc := createMockCloudDependencies(ctx, t, logger, validDeps)
+
+	cfg := resource.Config{ConvertedAttributes: replayCamCfg}
+	cam, err := newPCDCamera(ctx, resources, cfg, logger)
+
+	return cam, closeRPCFunc, err
+}
+
+// resourcesFromDeps returns a list of dependencies from the provided robot.
+func resourcesFromDeps(t *testing.T, r robot.Robot, deps []string) resource.Dependencies {
+	t.Helper()
+	resources := resource.Dependencies{}
+	for _, dep := range deps {
+		resName, err := resource.NewFromString(dep)
+		test.That(t, err, test.ShouldBeNil)
+		res, err := r.ResourceByName(resName)
+		if err == nil {
+			// some resources are weakly linked
+			resources[resName] = res
+		}
+	}
+	return resources
+}
+
+// getNextDataAfterFilter returns the next point cloud data to be return based on the provided filter and
+// last returned item.
```

Comment:
Nitpicky, but can we modify this comment to make it clear that this function just returns the index of the file?